src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -928,7 +928,10 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
     def migrationMessage     = getAnnotation(MigrationAnnotationClass) flatMap { _.stringArg(0) }
     def migrationVersion     = getAnnotation(MigrationAnnotationClass) flatMap { _.stringArg(1) }
     def elisionLevel         = getAnnotation(ElidableMethodClass) flatMap { _.intArg(0) }
-    def implicitNotFoundMsg  = getAnnotation(ImplicitNotFoundClass) flatMap { _.stringArg(0) }
+    def implicitNotFoundMsg  =
+      getAnnotation(ImplicitNotFoundClass).orElse(
```

Comment:
What I don't understand is that the failing tests are already using custom `implicitNotFound` messages, which presumably get returned by `getAnnotation(ImplicitNotFoundClass)`; if so, how is the fallback ever executed to produce a different result?

Regardless, I think it is worth adding a test for the correct message being used when there are multiple in the parent chain.

Reply:
There's a newish mechanism for checking if a param is annotated directly, before checking the type. "Parents of the param" accounts for seeing the wrong annotation (for some reason).